vividus-plugin-mobile-app/src/test/java/org/vividus/bdd/mobileapp/steps/NetworkConditionsStepsTests.java

```diff
@@ -0,0 +1,95 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.bdd.mobileapp.steps;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.vividus.mobileapp.action.NetworkActions;
+import org.vividus.mobileapp.action.NetworkActions.Mode;
+import org.vividus.mobileapp.action.NetworkActions.State;
+import org.vividus.selenium.manager.GenericWebDriverManager;
+
+@ExtendWith(MockitoExtension.class)
+class NetworkConditionsStepsTests
+{
+    @Mock
+    private GenericWebDriverManager genericWebDriverManager;
+    @Mock
+    private NetworkActions networkActions;
+    @InjectMocks
+    private NetworkConditionsSteps networkConditionsSteps;
+
+    @ParameterizedTest
+    @MethodSource("dataProvider")
+    void shouldFailForEnableNetworkConnectionForNotIOS(Mode mode, State state)
+    {
+        when(genericWebDriverManager.isIOS()).thenReturn(true);
+        IllegalArgumentException iae = assertThrows(IllegalArgumentException.class,
+                () -> networkConditionsSteps.changeNetworkConnection(mode, state));
+        assertEquals(String.format("%s is not supported for IOS", state), iae.getMessage());
+    }
+
+    @ParameterizedTest
+    @MethodSource("dataProviderForIOS")
+    void shouldEnableNetworkConnectionForIOS(Mode mode, State state)
+    {
+        when(genericWebDriverManager.isIOS()).thenReturn(true);
+        networkConditionsSteps.changeNetworkConnection(mode, state);
+        verify(networkActions).changeNetworkConnectionState(mode, state);
+    }
+
+    @ParameterizedTest
+    @MethodSource("dataProviderForAndroid")
+    void shouldEnableNetworkConnectionForAndroid(Mode mode, State state)
+    {
+        when(genericWebDriverManager.isAndroid()).thenReturn(true);
+        networkConditionsSteps.changeNetworkConnection(mode, state);
+        verify(networkActions).changeNetworkConnectionState(mode, state);
+    }
+
+    private static Stream<Arguments> dataProviderForAndroid()
+    {
+        return Stream.of(Arguments.of(Mode.DISABLE, State.WIFI), Arguments.of(Mode.ENABLE, State.WIFI),
+                Arguments.of(Mode.DISABLE, State.DATA), Arguments.of(Mode.ENABLE, State.DATA),
+                Arguments.of(Mode.DISABLE, State.AIRPLANE_MODE), Arguments.of(Mode.ENABLE, State.AIRPLANE_MODE),
+                Arguments.of(Mode.DISABLE, State.ALL), Arguments.of(Mode.ENABLE, State.ALL));
```

Comment:
```suggestion
        return Stream.of(
            Arguments.of(Mode.DISABLE, State.WIFI), 
            Arguments.of(Mode.ENABLE, State.WIFI),
            Arguments.of(Mode.DISABLE, State.DATA),
            Arguments.of(Mode.ENABLE, State.DATA),
            Arguments.of(Mode.DISABLE, State.AIRPLANE_MODE),
            Arguments.of(Mode.ENABLE, State.AIRPLANE_MODE),
            Arguments.of(Mode.DISABLE, State.ALL),
            Arguments.of(Mode.ENABLE, State.ALL)
        );
```
and the same approach for all other test data